packages/@tailwindcss-browser/src/index.ts

```diff
@@ -0,0 +1,295 @@
+
+// Warn users about using the browser build in production as early as possible.
+// It can take time for the script to do its work so this must be at the top.
+console.warn(
+  'The browser build of Tailwind CSS should not be used in production. To use Tailwind CSS in production, use the Tailwind CLI, Vite plugin, or PostCSS plugin: https://tailwindcss.com/docs/installation',
+)
+
+/**
+ * The type used by `<style>` tags that contain input CSS.
+ */
+const STYLE_TYPE = 'text/tailwindcss'
+
+/**
+ * The current Tailwind CSS compiler.
+ *
+ * This gets recreated:
+ * - When stylesheets change
+ */
+let compiler: Awaited<ReturnType<typeof tailwindcss.compile>>
+
+/**
+ * The list of all seen classes on the page so far. The compiler already has a
+ * cache of classes but this lets us only pass new classes to `build(…)`.
+ */
+let classes = new Set<string>()
+
+/**
+ * The last input CSS that was compiled. If stylesheets "change" without
+ * actually changing, we can avoid a full rebuild.
+ */
+let lastCss = ''
+
+/**
+ * The stylesheet that we use to inject the compiled CSS into the page.
+ */
+let sheet = document.createElement('style')
+
+/**
+ * The queue of build tasks that need to be run. This is used to ensure that we
+ * don't run multiple builds concurrently.
+ */
+let buildQueue = Promise.resolve()
+
+/**
+ * What build this is
+ */
+let nextBuildId = 1
+
+/**
+ * Used for instrumenting the build process. This data shows up in the
+ * performance tab of the browser's devtools.
+ */
+let I = new Instrumentation()
+
+/**
+ * Create the Tailwind CSS compiler
+ *
+ * This handles loading imports, plugins, configs, etc…
+ *
+ * This does **not** imply that the CSS is actually built. That happens in the
+ * `build` function and is a separate scheduled task.
+ */
+async function createCompiler() {
+  I.start(`Create compiler`)
+  I.start('Reading Stylesheets')
+
+  // The stylesheets may have changed causing a full rebuild so we'll need to
+  // gather the latest list of stylesheets.
+  let stylesheets: Iterable<HTMLStyleElement> = document.querySelectorAll(
+    `style[type="${STYLE_TYPE}"]`,
+  )
+
+  let css = ''
+  for (let sheet of stylesheets) {
+    observeSheet(sheet)
+    css += sheet.textContent + '\n'
+  }
+
+  // The user might have no stylesheets, or a some stylesheets without `@import`
+  // because they want to customize their theme so we'll inject the main import
+  // for them. However, if they start using `@import` we'll let them control
+  // the build completely.
+  if (!css.includes('@import')) {
+    css = `@import "tailwindcss";${css}`
+  }
+
+  I.end('Reading Stylesheets', {
+    size: css.length,
+    changed: lastCss !== css,
+  })
+
+  // The input CSS did not change so the compiler does not need to be recreated
+  if (lastCss === css) return
+
+  lastCss = css
+
+  I.start('Compile CSS')
+  try {
+    compiler = await tailwindcss.compile(css, {
+      base: '/',
+      async loadStylesheet(id, base) {
```

Comment:
This function doesn't use anything from outside its scope so we might want to move it into the module scope instead.

Reply:
Yeah can do that 👍